File: comunidata-api/src/main/java/com/senasoft/comunidataapi/chat/dto/response/ai/SimpleTextResponseDTO.java
package com.senasoft.comunidataapi.chat.dto.response.ai;

import com.senasoft.comunidataapi.chat.enums.ResponseType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class SimpleTextResponseDTO extends BaseDynamicResponseDTO {
    private String message;

    public SimpleTextResponseDTO(String summary, String analysis, String message) {
        super();
        this.setType(ResponseType.SIMPLE_TEXT);
        this.setSummary(summary);
        this.setAnalysis(analysis);
        this.message = message;
    }
}

File: comunidata-ui/lib/dto/chat/response/simple_text_response_dto.dart
import 'base_dynamic_response_dto.dart';

class SimpleTextResponseDTO extends BaseDynamicResponseDTO {
  final String text;

  const SimpleTextResponseDTO({
    required super.type,
    super.summary,
    super.analysis,
    required this.text,
  });

  factory SimpleTextResponseDTO.fromJson(Map<String, dynamic> json) {
    return SimpleTextResponseDTO(
      type: ResponseType.SIMPLE_TEXT,
      summary: json['summary'],
      analysis: json['analysis'],
      text: json['text'] ?? '',
    );
  }

  @override
  Map<String, dynamic> toJson() => {
        'type': type.name,
        'summary': summary,
        'analysis': analysis,
        'text': text,
      };
}
